crates/router/src/core/payment_methods.rs

```diff
@@ -2650,3 +2667,169 @@ impl pm_types::SavedPMLPaymentsInfo {
         Ok(())
     }
 }
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+#[allow(clippy::too_many_arguments)]
+async fn add_token_call_to_store(
+    state: SessionState,
+    req_state: routes::app::ReqState,
+    merchant_account: domain::MerchantAccount,
+    profile: domain::Profile,
+    key_store: domain::MerchantKeyStore,
+    payment_method_create_request: &payment_methods::PaymentMethodCreate,
+    payment_method: &api::PaymentMethodResponse,
+    payment_method_session: &domain::payment_methods::PaymentMethodSession,
+) -> RouterResult<()> {
+    use crate::{db::errors::ConnectorErrorExt, types::Tokenizable};
+
+    let customer_id = payment_method_create_request.customer_id.to_owned();
+    let connector_id = payment_method_create_request
+        .psp_tokenization
+        .clone()
+        .get_required_value("psp_tokenization")?
+        .connector_id
+        .get_required_value("connector_id")?;
+    let db = &state.store;
+    // call merchant connector account via merchant_connector_id ( connector, connector_auth_type)
+    let merchant_connector_account_details = db
+        .find_merchant_connector_account_by_id(&(&state).into(), &connector_id, &key_store)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
+            id: connector_id.get_string_repr().to_owned(),
+        })
+        .attach_printable("error while fetching merchant_connector_account from connector_id")?;
+    let auth_type = merchant_connector_account_details
+        .get_connector_account_details()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed while parsing value for ConnectorAuthType")?;
+
+    let payment_method_data_request = types::PaymentMethodTokenizationData {
+        payment_method_data: payment_method_data::PaymentMethodData::try_from(
+            payment_method_create_request.payment_method_data.clone(),
+        )
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to convert type from ")?,
+        browser_info: None,
+        currency: api_models::enums::Currency::default(),
+        amount: None,
+    };
+
+    let payment_method_session_address = types::PaymentAddress::new(
+        None,
+        payment_method_session
+            .billing
+            .clone()
+            .map(|address| address.into_inner()),
+        None,
+        None,
+    );
+
+    let mut router_data =
+        types::RouterData::<api::PaymentMethodToken, _, types::PaymentsResponseData> {
```

Comment:
can we use RouterDatav2 here and can we make a function instead of building the object here, another way is we can create builder pattern for RouterData and add data to builder and build the router_data out of it.